### ISSUE TITLE
Are data-rel="dialog" and #ui-state="dialog" currently supported?

### ISSUE THREAD

**Issue by blasky** [opitzconsulting/jquery-mobile-angular-adapter#83]
This first fiddle has a main page, a dialog page, and a couple of buttons that enable you to transition back and forth. The adapter is not present in this fiddle, just jQuery Mobile doing what it wants to do with a dialog transition.

   http://jsfiddle.net/tuHxw/3/

This second fiddle is the exact same structure with the angular adapter code added in.

   http://jsfiddle.net/FHVMh/3/

The "Show Url" links on each page can be clicked at any point to alert the value of the "document.location" within the fiddle preview frame (...since you can't see the url for the pages otherwise).

In the Mobile only example, the URL transtions back and forth between "http://fiddle.jshell.net/_display/..." and "http://fiddle.jshell.net/_display/...#&ui-state=dialog". And the dialog behaves as it does on the Mobile example pages.

After the adapter has taken over, the URL starts at "http://fiddle.jshell.net/_display/..." and then transitions between "http://fiddle.jshell.net/_display/...#dialogPage" and "http://fiddle.jshell.net/_display/...#mainPage".

(The dialog is also no longer being rendered as a dialog in the adapter fiddle, but that's another issue.)

So, in a nutshell, it appears that after the adapter code runs the page transitions are ignoring the "data-rel" attributes that have been set on the buttons and the special "ui-state" behavior is nowhere to be found.

This creates strange behavior in the history and routing. And more importantly, other controls which rely upon a dialog with "ui-state" based behavior (...such as a custom menu) are failing in other ways.

**Comment by tbosch** [comment]
Hi,
thanks for reporting this.
The problem with the urls/routing is a known issue of the current snapshot release. You could either use the 1.1.1 version together with jquery mobile 1.1.1, or wait for the release of the 1.1.2 version. This is due to the internal refactorings of the routing mechanism, as we now use the routing of angularjs. This issue is related to #70 which I will fix next.

The other problem with not respecting the ui-state attributes is another one. Could you try again with the 1.1.1 version of the adapter, together with jquery mobile 1.1.1?

Thanks,
Tobias

**Comment by blasky** [comment]
And thank you for the quick response. I'm sure it's tricky having to develop right in the middle of the two libraries and having to juggle changes on both sides as angular and mobile continue to shift and change beneath you.

I will experiment with 1.1.1 and see what the trade offs look like (...known bugs from the previous releases, unavailable features, etc.). Do you have a general time frame you're shooting for on 1.1.2?

Thanks.

**Comment by tbosch** [comment]
My goal is before christmas!

**Comment by tbosch** [comment]
Hi,
this should work now, see this jsfiddle: http://jsfiddle.net/FHVMh/6/

Please note that I moved the "ng-app" attribute up to the body (See the info section in the jsfiddle) so that both pages are within angular!

One point that does not work is using links with `data-rel="dialog"`, e.g. `<a href="somePage" data-rel="dialog">`. Instead, please use `<div data-role="dialog">` on the dialog page. 

Tobias

**Comment by blasky** [comment]
The fiddle looks great. I will start verifying on my project code. Thanks.
